app/models/user.rb

```diff
@@ -28,7 +20,7 @@ def set_location
   # Include default devise modules. Others available are:
   # :omniauthable
   devise :database_authenticatable, :registerable, :confirmable, :timeoutable,
-         :recoverable, :rememberable, :trackable, :validatable, :lockable
```

Comment:
Why do this with manual validation instead of using the devise validatable? 
This could be a problem, at least we need to put a comment on the devise initializer.